pennylane/qchem/openfermion_obs.py

```diff
@@ -1077,11 +1068,14 @@ def _molecular_hamiltonian(
             raise ValueError(
                 "Only 'jordan_wigner' mapping is supported for the differentiable workflow."
             )
-        if mult != 1:
+        n_electrons = sum([atomic_numbers[s] for s in symbols]) - charge
+
+        if n_electrons % 2 == 1 or mult != 1:
             raise ValueError(
-                "Openshell systems are not supported for the differentiable workflow. Use "
+                "Open-shell systems are not supported for the differentiable workflow. Use "
                 "`method = 'pyscf'` or change the charge or spin multiplicity of the molecule."
             )
+
```

Comment:
Please change this line to:
"Open-shell systems are not supported for the requested backend. Use "`method = 'openfermion'` or change the charge or spin multiplicity of the molecule."